dspace-server-webapp/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java

```diff
@@ -58,16 +77,45 @@ protected void doFilterInternal(HttpServletRequest req,
                                     HttpServletResponse res,
                                     FilterChain chain) throws IOException, ServletException {
 
-        Authentication authentication = getAuthentication(req);
+        Authentication authentication = null;
+        try {
+            authentication = getAuthentication(req, res);
+        } catch (AuthorizeException e) {
+            res.sendError(HttpServletResponse.SC_UNAUTHORIZED, e.getMessage());
+            log.error(e.getMessage(), e);
+            return;
+        } catch (IllegalArgumentException | SQLException e) {
+            res.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+            log.error(e.getMessage(), e);
+            return;
+        } catch (AccessDeniedException e) {
+            res.sendError(HttpServletResponse.SC_FORBIDDEN, e.getMessage());
+            log.error(e.getMessage(), e);
+            return;
+        }
         if (authentication != null) {
             SecurityContextHolder.getContext().setAuthentication(authentication);
             restAuthenticationService.invalidateAuthenticationCookie(res);
         }
-
         chain.doFilter(req, res);
     }
 
-    private Authentication getAuthentication(HttpServletRequest request) {
+    /**
+     * This method will return a Pair instance with an Authentication object as the left side of the pair and a Boolean
+     * for the right side of the pair which will indicate whether there was an error in the OnBehalfOf parsing or not
+     * The Authentication object will be attempted to be for the eperson with the uuid in the parameter. Incase
+     * this is able to be done properly, we'll be giving a pair back with the EPerson Authentication in the left side
+     * and a false boolean as the right side.
+     * If the Authentication object returned is not null, we'll be logged in as this EPerson given through from the
+     * request. If the Boolean is true, we'll stop the execution and show a BadRequest error
+     * @param request       The current request
+     * @param res           The current response
+     * @return              A Pair instance with the Authentication object on the left side and the boolean
+     *                      indicating errors on the right side
+     * @throws IOException  If something goes wrong
+     */
+    private Authentication getAuthentication(HttpServletRequest request, HttpServletResponse res)
```

Comment:
These JavaDocs are now incorrect as they describe how the method worked prior to the refactor. The `Pair` was removed.  @KevinVdV could we get these updated quickly?  This is the last thing I see that needs updating before we can merge.